src/cluster.c

```diff
@@ -1613,15 +1613,13 @@ unsigned int clusterDelKeysInSlot(unsigned int hashslot, int flags) {
         enterExecutionUnit(1, 0);
         sds sdskey = kvobjGetKey(dictGetKV(de));
         robj *key = createStringObject(sdskey, sdslen(sdskey));
-
-        if (flags & CLUSTER_DELKEYS_ASYNC) dbAsyncDelete(&server.db[0], key);
-        else dbSyncDelete(&server.db[0], key);
+        dbSyncDelete(&server.db[0], key);
```

Comment:
reverting the half baked async delete here in this function. We won't rely on this for asm anyway.